src/decompose.rs

```diff
@@ -10,7 +10,10 @@
 use core::fmt::{self, Write};
 use core::iter::Fuse;
 use core::ops::Range;
-use tinyvec::TinyVec;
+#[cfg(feature = "alloc")]
+type VecType<T> = tinyvec::TinyVec::<[T; 4]>;
+#[cfg(not(feature = "alloc"))]
+type VecType<T> = tinyvec::ArrayVec::<[T; 128]>;
```

Comment:
Could you document the reasoning for choosing `4` and `128` here? I can guess (4 is more common, 128 to accommodate most use-cases when the heap is not available) but perhaps we should be explicit?

Reply:
My choices here were arbitrary and I don't know if this is what these numbers should be. I just took a guess. However if people think these numbers are good I'm also happy to document why we, collectively, think they're good.

Reply:
Well the 4 is never going to panic, however I did do some amount of testing for the 128. The unit test `test_streamsafe_regression' will fail if the value is 32, and it won't fail if the value is 64. So to be very safe, I doubled the one that didn't fail.

Reply:
Would 256 be more tenable? Using 64 instead won't cause the unit tests to fail.